internal/contract/controlplane.go

```diff
@@ -351,3 +353,58 @@ func (c *ControlPlaneMachineTemplate) NodeDeletionTimeout() *Duration {
 		path: Path{"spec", "machineTemplate", "nodeDeletionTimeout"},
 	}
 }
+
+// ReadinessGates provides access to control plane's ReadinessGates.
+func (c *ControlPlaneContract) ReadinessGates() *ReadinessGates {
+	return &ReadinessGates{}
+}
+
+// ReadinessGates provides a helper struct for working with ReadinessGates.
+type ReadinessGates struct{}
+
+// Path returns the path of the ReadinessGates.
+func (m *ReadinessGates) Path() Path {
+	return Path{"spec", "machineTemplate", "readinessGates"}
+}
+
+// Get gets the ReadinessGates object.
+func (m *ReadinessGates) Get(obj *unstructured.Unstructured) ([]clusterv1.MachineReadinessGate, error) {
+	unstructuredValue, ok, err := unstructured.NestedSlice(obj.UnstructuredContent(), m.Path()...)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve control plane %s", m.Path())
+	}
+	if !ok {
+		return nil, nil
+	}
+	jsonValue, err := json.Marshal(&unstructuredValue)
```

Comment:
```suggestion
	jsonValue, err := json.Marshal(unstructuredValue)
```
Hm, this is what we do for MachineAddresses if I see correctly. Do you know why we do this differently here and if both works the same way?

In general the both funcs are implemented pretty differently considering they are doing ~ the same thing (e.g. we are also using a leading `.` in the paths of MachineAddresses)